### PR TITLE
Add required resources locally for offline html report generation of Ballerina profiler

### LINKED ISSUE #41427: [Bug]: Profiler output does not show properly when you are offilne
### Description

When we do not have internet, profile output html file does not show properly.

### Steps to Reproduce

_No response_

### Affected Version(s)

_No response_

### OS, DB, other environment details and versions

_No response_

### Related area

-> Runtime

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by gabilang**
Fixes #41427

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-profiler/src/main/java/io/ballerina/runtime/profiler/ui/HttpServer.java [hunk 1]
```diff
@@ -33,17 +44,50 @@
...
+            copyFolder(resourcePath, Path.of(profilerOutputDir));
+        } catch (IOException e) {
+            OUT_STREAM.printf("%s%n", e);
+        }
+
         String content = FileUtils.readFileAsString("performance_report.json");
         FrontEnd frontEnd = new FrontEnd();
         String htmlData = frontEnd.getSiteData(content);
-        String fileName = "ProfilerOutput.html";
+        String fileName = "ProfilerReport.html";
```
Comment (HindujaB) [inline bvm/ballerina-profiler/src/main/java/io/ballerina/runtime/profiler/ui/HttpServer.java:63]: We can move `ProfilerReport.html` to the `Constants` class and reuse it.

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunProfilerTask.java [hunk 2]
```diff
@@ -125,14 +109,31 @@ private String getAgentArgs() {
...
             return Paths.get(System.getProperty(USER_DIR));
         }
-        return project.targetDir().resolve("bin");
+        return project.targetDir();
+    }
+
+    private String getTargetFilePath(Project project) {
+        if (project.kind() == ProjectKind.SINGLE_FILE_PROJECT) {
+            return Paths.get(System.getProperty(USER_DIR)).resolve(
+                    getFileNameWithoutExtension(project.sourceRoot()) + BLANG_COMPILED_JAR_EXT).toUri().getPath();
```
Comment (HindujaB) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunProfilerTask.java:134]: Shall we check why this line is not covered?

File: bvm/ballerina-profiler/src/main/java/io/ballerina/runtime/profiler/ui/HttpServer.java [hunk 3]
```diff
@@ -33,17 +44,50 @@
...
         try (FileWriter writer = new FileWriter(fileName, StandardCharsets.UTF_8)) {
             writer.write(htmlData);
         } catch (IOException e) {
             OUT_STREAM.printf("%s%n", e);
         }
     }
+
+    public static void copyFolder(Path source, Path target) throws IOException {
```
Comment (warunalakshitha) [inline bvm/ballerina-profiler/src/main/java/io/ballerina/runtime/profiler/ui/HttpServer.java:71]: Can be private

File: bvm/ballerina-profiler/src/main/java/io/ballerina/runtime/profiler/ui/HttpServer.java [hunk 4]
```diff
@@ -33,17 +44,50 @@
...
         OUT_STREAM.printf(" ○ Output: " + Constants.ANSI_YELLOW +
-                "%s/ProfilerOutput.html" + Constants.ANSI_RESET + "%n", sourceRoot);
+                "%s/ProfilerReport.html" + Constants.ANSI_RESET + "%n", profilerOutputDir);
+        Path resourcePath = Paths.get(System.getenv(BALLERINA_HOME)).resolve("resources")
+                .resolve("profiler");
+
+        try {
+            copyFolder(resourcePath, Path.of(profilerOutputDir));
+        } catch (IOException e) {
+            OUT_STREAM.printf("%s%n", e);
```
Comment (warunalakshitha) [inline bvm/ballerina-profiler/src/main/java/io/ballerina/runtime/profiler/ui/HttpServer.java:57]: Shall we throw ballerina Profiler exception

File: tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/profiler/ProfilerTest.java [hunk 5]
```diff
@@ -41,8 +41,9 @@
 public class ProfilerTest extends BaseTest {
     private static final String testFileLocation = Paths.get("src/test/resources/profiler")
```
Comment (warunalakshitha) [inline tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/profiler/ProfilerTest.java:42]: Shall we use here Paths.get(src, test, resources,profiler). test will not work in windows

File: tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/profiler/ProfilerTest.java [hunk 6]
```diff
@@ -51,11 +52,11 @@ public void setup() throws BallerinaTestException {
 
     @Test
     public void testProfilerExecutionWithBalPackage() throws BallerinaTestException {
-        String sourceRoot = testFileLocation + "/";
         String packageName = "projectForProfile/package_a";
+        String sourceRoot = testFileLocation + "/";
         Map<String, String> envProperties = new HashMap<>();
         bMainInstance.addJavaAgents(envProperties);
-        LogLeecher[] leechers = getProfilerLogLeechers(packageName + "/target/bin/" + outputFile);
```
Comment (warunalakshitha) [inline tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/profiler/ProfilerTest.java:58]: Shall we use paths api

File: tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/profiler/ProfilerTest.java [hunk 7]
```diff
@@ -51,11 +52,11 @@ public void setup() throws BallerinaTestException {
 
     @Test
     public void testProfilerExecutionWithBalPackage() throws BallerinaTestException {
-        String sourceRoot = testFileLocation + "/";
         String packageName = "projectForProfile/package_a";
+        String sourceRoot = testFileLocation + "/";
```
Comment (warunalakshitha) [inline tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/profiler/ProfilerTest.java:56]: We can use file.seperator here

File: tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/profiler/ProfilerTest.java [hunk 8]
```diff
@@ -84,7 +85,8 @@ public void testProfilerExecutionWithSingleBalFile() throws BallerinaTestExcepti
         String fileName = "profiler_single_file.bal";
         Map<String, String> envProperties = new HashMap<>();
         bMainInstance.addJavaAgents(envProperties);
-        LogLeecher[] leechers = getProfilerLogLeechers(sourceRoot + outputFile);
+        envProperties.put(BALLERINA_HOME, bMainInstance.getBalServerHome());
+        LogLeecher[] leechers = getProfilerLogLeechers(sourceRoot + "profiler/" + outputFile);
```
Comment (warunalakshitha) [inline tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/profiler/ProfilerTest.java:89]: Lets use Paths.get or file.seperator

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunProfilerTask.java (hunk 2 above)
Comment (gabilang) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunProfilerTask.java:136]: When running the added single file test locally, those lines are hit.

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**